dali/kernels/imgproc/pointwise/linear_transformation_gpu_test.cu

```diff
@@ -0,0 +1,242 @@
+  TheKernel<TypeParam> kernel;
+  KernelContext ctx;
+  InListGPU<typename TypeParam::In, kNDims> in(this->input_device_, this->in_shapes_);
+  auto reqs = kernel.Setup(ctx, in, make_cspan(this->vmat_), make_cspan(this->vvec_),
+                           make_cspan(this->rois_));
+  auto ref_shape = ShapeFromRoi(this->rois_[0], kNChannelsOut);
+  ASSERT_EQ(ref_shape, reqs.output_shapes[0][0]);
+}
+
+
+TYPED_TEST(LinearTransformationGpuTest, run_test) {
+  TheKernel<TypeParam> kernel;
+  KernelContext c;
+  InListGPU<typename TypeParam::In, kNDims> in(this->input_device_, this->in_shapes_);
+
+  auto reqs = kernel.Setup(c, in, make_cspan(this->vmat_), make_cspan(this->vvec_));
+
+  ScratchpadAllocator sa;
+  sa.Reserve(reqs.scratch_sizes);
+  auto scratchpad = sa.GetScratchpad();
+  c.scratchpad = &scratchpad;
+
+  OutListGPU<typename TypeParam::Out, kNDims> out(
+          this->output_, reqs.output_shapes[0].template to_static<kNDims>());
+
+  kernel.Run(c, out, in, make_cspan(this->vmat_), make_cspan(this->vvec_));
+  cudaDeviceSynchronize();
+
+  auto res = copy<AllocType::Host>(out[0]);
+  ASSERT_EQ(static_cast<int>(this->ref_output_.size()), res.first.num_elements());
+  for (size_t i = 0; i < this->ref_output_.size(); i++) {
+    EXPECT_PRED3((IsEqWithConvert<typename TypeParam::Out, float>), res.second.get()[i],
+                 this->ref_output_[i], 4) << "Failed at idx: " << i;
+  }
+}
+
+
+TYPED_TEST(LinearTransformationGpuTest, run_test_with_roi) {
+  TheKernel<TypeParam> kernel;
+  KernelContext c;
+  InListGPU<typename TypeParam::In, kNDims> in(this->input_device_, this->in_shapes_);
+
+  auto reqs = kernel.Setup(c, in,
+                           make_cspan(this->vmat_), make_cspan(this->vvec_),
+                           make_cspan(this->rois_));
+
+  ScratchpadAllocator sa;
+  sa.Reserve(reqs.scratch_sizes);
+  auto scratchpad = sa.GetScratchpad();
+  c.scratchpad = &scratchpad;
+
+  OutListGPU<typename TypeParam::Out, kNDims> out(
+          this->output_, reqs.output_shapes[0].template to_static<kNDims>());
+
+  kernel.Run(c, out, in, make_cspan(this->vmat_), make_cspan(this->vvec_), make_cspan(this->rois_));
+  cudaDeviceSynchronize();
+
+  auto res = copy<AllocType::Host>(out[0]);
+  auto mat = color_manipulation::test::to_mat<kNChannelsOut>(this->ref_output_.data(),
+                                                             this->rois_[0],
+                                                             this->out_shapes_[0][0],
+                                                             this->out_shapes_[0][1]);
+
+  ASSERT_EQ(mat.rows * mat.cols * mat.channels(), res.first.num_elements())
+                        << "Number of elements doesn't match";
+  auto ptr = reinterpret_cast<typename TypeParam::Out *>(mat.data);
+  for (int i = 0; i < res.first.num_elements(); i++) {
+    EXPECT_PRED3((IsEqWithConvert<typename TypeParam::Out, float>), res.second.get()[i], ptr[i], 4)
```

Comment:
Nice :+1:! I didn't know about these macros! It would be great to replace EXPECT_TRUE with EXPECT_PRED2 in Check function.

Reply:
Done